### PR TITLE
Added Tool Entity, repository and tests

### PR DESCRIPTION
**Pull request by it-rookie**
Please review my changes

### CODE REVIEW

File: src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+        then:
+        response.status == HttpStatus.CREATED.value()
+        response.headers.'Content-Type'.toString() == 'application/hal+json;charset=UTF-8'
+        String location = response.headers.'Location'.toString()
+        location.startsWith('http://localhost:8080/tool/')
+    }
+
+    def 'get to /tool/id read tool'() {
+        given:
```
Comment (jenac) [inline src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:55]: sample question

File: src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,162 @@
+package com.mnit.tnt.repository
+import groovy.json.JsonSlurper
```
Comment (jenac) [inline src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:2]: question 1?

Comment (edenprairie) [reply src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:2]: I think this is good.

Comment (it-rookie) [reply src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:2]: Answer 1.

File: src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy (hunk 1 above)
Comment (xwowenzhao) [reply src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:55]: xiwen test comment

File: src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy (hunk 2 above)
Comment (edenprairie) [reply src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:2]: I will try to use start review

File: src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy (hunk 1 above)
Comment (it-rookie) [reply src/integration/java/com/mnit/tnt/repository/ToolRestApiIntegrationTest.groovy:55]: Sample answer

**Review by edenprairie — COMMENTED** [review]
comment for start review.

**Pull request merged**